tests/desktop/test_users_account.py

```diff
@@ -86,13 +86,12 @@ def test_hide_email_checkbox_works(self, mozwebqa):
         initial_state = view_profile_page.is_email_field_present
 
         edit_profile_page = home_page.header.click_edit_profile()
-        edit_profile_page.change_hide_email_state()
+        edit_profile_page.hide_email_address()
         edit_profile_page.click_update_account()
-
         view_profile_page = home_page.header.click_view_profile()
         final_state = view_profile_page.is_email_field_present
 
-        Assert.not_equal(initial_state, final_state, 'The initial and final states are the same. The profile change failed.')
+        Assert.equal(view_profile_page.is_email_field_present, False, 'The email address is not hidden')
```

Comment:
There is a problem with the test. If you run it twice it will fail (at least it does for me). The first time it clicks the box, and then the second time it will find the box already selected. Perhaps the test should both hide and then reshow the email address. First it could hide it, by checking the box, and then verify that the email address is hidden, and then it could show it again, by unchecking the box (which is where you'd make use of `show_email_address`) and then verify that it appears again.

There is still an issue that we are using an existing user and therefore cannot be sure of the initial state of the checkbox. If all goes well with our test, it should be in an unchecked state at the end of the test, but if the test fails or errors partway through then it might leave the user in a state where the checkbox is checked, in which case the next run of the test will fail.

I don't suppose we have the option of using a new user for each run of the test?

Another option would be to have a setup step where we make sure that the box is unchecked, and if it is not then we uncheck it.

Do you follow what I'm saying above? What do you think?